### ISSUE TITLE
How to combine Object Detection YOLO and Image Classification SVM models?

### ISSUE THREAD

**Issue by MuhammadFikri1903** [ultralytics/ultralytics#19973]
Hello all, I am currently confused about the right steps in combining yolo detection objects with classification using the svm model. is it necessary to do feature extraction first using the transfer learning algorithm before using svm for classification? 
Thank you.

![Image](https://github.com/user-attachments/assets/5b03fe93-588d-4405-9323-58b8d7d4b0e5)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @MuhammadFikri1903, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for guidance, where you can find detailed information and examples for various tasks. This might help clarify some of your questions.

If this is a ❓ Question, please provide as much detail as possible, such as any specific implementation steps you’ve tried, and ensure you're following the best practices outlined in our [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/).

If you’re encountering a 🐛 Bug, kindly include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that outlines your workflow. This will help us investigate the issue effectively.

To further explore your use case, feel free to join the Ultralytics community:
- 💬 [Discord](https://discord.com/invite/ultralytics) for real-time chat
- 📚 [Discourse](https://community.ultralytics.com/) for detailed discussions
- 🔗 [Subreddit](https://reddit.com/r/Ultralytics) for insights from the community

## Upgrade

Ensure you're using the latest `ultralytics` package, which includes improvements and fixes. Update it in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in verified environments with dependencies preinstalled:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **AWS**: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests validate all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

This is an automated response to guide you 🌟. An Ultralytics engineer will review your issue and assist you further soon!

**Comment by glenn-jocher** [comment]
@MuhammadFikri1903 to combine YOLO object detection with SVM classification: 

1. Use YOLO to detect and crop objects from images
2. Extract features from cropped regions (YOLO's backbone can serve as feature extractor - no transfer learning required unless you want to fine-tune it)
3. Train SVM on these features

For implementation:
```python
from ultralytics import YOLO

# Detect objects
model = YOLO('yolov8n.pt')
results = model('image.jpg')

# Crop detections and extract features
for box in results[0].boxes.xyxy:
    crop = im.crop(box.cpu().numpy())
    features = model.backbone(crop)  # Extract features using YOLO's backbone
    # Train/predict with SVM using features
```

While traditional approaches used HOG+SVM (see [object detection evolution](https://www.ultralytics.com/blog/the-evolution-of-object-detection-and-ultralytics-yolo-models)), modern implementations often use YOLO's native classification head instead of SVM for end-to-end training. Consider trying `YOLO('yolov8n-cls.pt')` for classification tasks.

**Comment by MuhammadFikri1903** [comment]
> [@MuhammadFikri1903](https://github.com/MuhammadFikri1903) to combine YOLO object detection with SVM classification:
> 
> 1. Use YOLO to detect and crop objects from images
> 2. Extract features from cropped regions (YOLO's backbone can serve as feature extractor - no transfer learning required unless you want to fine-tune it)
> 3. Train SVM on these features
> 
> For implementation:
> 
> from ultralytics import YOLO
> 
> # Detect objects
> model = YOLO('yolov8n.pt')
> results = model('image.jpg')
> 
> # Crop detections and extract features
> for box in results[0].boxes.xyxy:
>     crop = im.crop(box.cpu().numpy())
>     features = model.backbone(crop)  # Extract features using YOLO's backbone
>     # Train/predict with SVM using features
> While traditional approaches used HOG+SVM (see [object detection evolution](https://www.ultralytics.com/blog/the-evolution-of-object-detection-and-ultralytics-yolo-models)), modern implementations often use YOLO's native classification head instead of SVM for end-to-end training. Consider trying `YOLO('yolov8n-cls.pt')` for classification tasks.

Thank you very much for your response and explanation! Your explanation was very clear and opened new insights for me about using YOLO backbone for feature extraction. I really appreciate your help and the time you took to answer my questions. 🙏

**Comment by glenn-jocher** [comment]
You're welcome, @MuhammadFikri1903! Happy to help. If you encounter any challenges implementing either the YOLO+SVM approach or the end-to-end YOLO classification, feel free to ask further questions. Good luck with your project!
